src/utils/functionRuntimeUtils.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as opn from 'opn';
+import * as vscode from 'vscode';
+import { extensionPrefix } from '../../src/ProjectSettings';
+import { DialogResponses } from '../DialogResponses';
+import { localize } from '../localize';
+import { cpUtils } from './cpUtils';
+import { nodeUtils } from './nodeUtils';
+
+export namespace functionRuntimeUtils {
+
+    const runtimePackage: string = 'azure-functions-core-tools';
+
+    export async function validateFunctionRuntime(outputChannel: vscode.OutputChannel): Promise<void> {
+        const errorMessage: string | undefined = await getRuntimeValidationError(outputChannel);
+        if (errorMessage) {
+            await promptDocumentationAction(errorMessage);
+        }
+    }
+
+    async function getRuntimeValidationError(outputChannel: vscode.OutputChannel): Promise<string | undefined> {
+        try {
+            await cpUtils.executeCommand(outputChannel, undefined, 'func');
+        } catch (error) {
+            return localize(
+                'azFunc.noFunctionRuntime',
+                '[Azure Functions] Azure-functions-core-tools is not installed. Please install it otherwise Azure Functions extension may fail.'
+            );
+        }
+
+        try {
+            await nodeUtils.validateNpmInstalled(outputChannel);
+        } catch (error) {
+            return localize(
+                'azFunc.npmNotFound',
+                '[Azure Functions] Failed to check function runtime version caused by "npm" not in path, would you like to check the document for the prerequisites?'
+            );
+        }
+
+        const outdatedInfo: string = await cpUtils.executeCommand(outputChannel, undefined, 'npm', 'outdated', runtimePackage, '-g');
```

Comment:
This doesn't work for me. I'm on the latest version, but it tells me to install a version from back in September
```
Package                           Current          Wanted  Latest  Location
azure-functions-core-tools  2.0.1-beta.22  2.0.1-beta1.15   1.0.8
```

I think you want a combination of these commands:
`npm view azure-functions-core-tools dist-tags` <- To get the latest supported for both version 1.0 and 2.0
`npm ls azure-functions-core-tools -g` <- To get the version on the machine